Quartz/Engine/Include/Quartz/Utilities/Commander.hpp

```diff
@@ -0,0 +1,143 @@
+// Copyright 2019 Genten Studios
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation
+// and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#pragma once
+
+#include <array>
+#include <functional>
+#include <string>
+
+namespace qz
+{
+	namespace utils
+	{
+		constexpr std::size_t MAX_COMMANDS_NUMBER  = 100;
+		constexpr std::size_t MAX_ARGUMENTS_NUMBER = 10;
+		typedef std::function<int(
+		    std::array<std::string, MAX_ARGUMENTS_NUMBER> args)>
+		    function;
+
+		/**
+		 * @brief The command book stores commands and information on them to be
+		 * used by a commander
+		 */
+
+		struct CommandBook
+		{
+			std::array<std::string, MAX_COMMANDS_NUMBER> m_command;
+			std::array<std::string, MAX_COMMANDS_NUMBER> m_help;
+			std::array<std::string, MAX_COMMANDS_NUMBER> m_permission;
+			std::array<function, MAX_COMMANDS_NUMBER>    m_functions;
+
+			/**
+			 * @brief Registers a command in the command registry
+			 *
+			 * @param command The keyword for calling the command
+			 * @param permission What permission is required to run this command
+			 */
+			void add(const std::string& command, const std::string& help,
+			         const std::string& permission, function f);
+
+			/**
+			 * @brief Searches for a command
+			 *
+			 * @param command The command to search for
+			 * @return index command is located at or -1 if its not found
+			 */
+			int find(const std::string& command);
+
+			/**
+			 * @brief Gets next open space in book
+			 */
+			int getPage();
```

Comment:
If you could just documented expected type/range of return values that could be expected from this function